policy engine should 400 (with instructive message) on invalid bundle upload instead of 500

<!-- Thanks for trying Anchore Engine and filing an issue! To help ensure that we can quickly and efficiently process issues and get fixes please answer a few questions-->

**Is this a request for help?**: NO

---

**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one): BUG REPORT

<!--
If this is a BUG REPORT, please:
  - Fill in as much of the template below as possible so we can reproduce and understand the issue.

If this is a FEATURE REQUEST, please:
  - Describe the feature *in detail* as much as possible including possible cli and api behavior and how to test the feature.

In both cases, we'll probably ask follow-up questions to ensure it's well understood. If we cannot reproduce the issue or
questions go unanswered for too long, we may close the issue. If we're wrong, PLEASE feel free to re-open it and explain why.
-->

<!-- To get version info:
anchore-cli --version
docker exec <engine_container> pip show anchore-engine
-->
**Version of Anchore Engine and Anchore CLI if applicable**: Engine/CLI 0.7.2 & Enterprise/UI 2.3.1


**What happened**: Tried to upload a policy bundle through the UI & CLI that had policy_ids in some mappings but I hadn't actually created the policies


**What did you expect to happen**: To be presented with a more meaningful error

<!--
Log output can be from the container stdout or /var/log/anchore/anchore-*.
The /var/log/anchore/anchore-* logs are service specific and can help identify which components is having the issue, so please check
there as well as the stdout from the container.
-->
**Any relevant log output from /var/log/anchore**:

1. Error message: 
`Policy bundle addition returned a status code of 500 with the message: "Error response from policy service during bundle validation. Validation could not be performed: failed post url=http://anchore-anchore-engine-policy:8087/v1/validate_bundle"`
2. policy engine pod logs:
`[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-] Traceback (most recent call last):
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]   File "/usr/local/lib/python3.6/site-packages/anchore_engine/services/policy_engine/api/controllers/synchronous_operations.py", line 807, in validate_bundle
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]     resp.validation_details = [problem_from_exception(i, severity='error') for i in problems]
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]   File "/usr/local/lib/python3.6/site-packages/anchore_engine/services/policy_engine/api/controllers/synchronous_operations.py", line 807, in <listcomp>
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]     resp.validation_details = [problem_from_exception(i, severity='error') for i in problems]
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]   File "/usr/local/lib/python3.6/site-packages/anchore_engine/services/policy_engine/api/controllers/synchronous_operations.py", line 233, in problem_from_exception
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-]     prob.details = eval_exception.message
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-] AttributeError: 'AttributeError' object has no attribute 'message'
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [_GenericHTTPChannelProtocol,318800,10.131.6.222] [PoolThread-twisted.internet.reactor-0] [anchore_engine.services.policy_engine.api.controllers.synchronous_operations/validate_bundle()] [ERROR] Failed processing bundle evaluation: 'AttributeError' object has no attribute 'message'
[service:enterprise-policy-engine] 2020-10-01 16:46:55+0000 [-] "10.131.6.222" - - [01/Oct/2020:16:46:55 +0000] "POST /v1/validate_bundle HTTP/1.1" 500 130 "-" "python-requests/2.23.0"`


**What docker images are you using**: Anchore Enterprise, UI, & CLI


**How to reproduce the issue**: Upload a bundle with a mapping that specifies a policy_id but doesn't actually have the policy included in the bundle


**Anything else we need to know**: YOU GUYS ROCK! 😎

--- a/anchore_engine/services/policy_engine/api/controllers/synchronous_operations.py
+++ b/anchore_engine/services/policy_engine/api/controllers/synchronous_operations.py
@@ -229,8 +229,10 @@ def problem_from_exception(eval_exception, severity=None):
     # If there is a details() function, call that
     if hasattr(eval_exception, 'details') and callable(eval_exception.details):
         prob.details = eval_exception.details()
-    else:
+    elif hasattr(eval_exception, 'message'):
         prob.details = eval_exception.message
+    else:
+        prob.details = str(eval_exception)
 
     prob.problem_type = eval_exception.__class__.__name__
     if hasattr(eval_exception, 'severity') and eval_exception.severity:

--- a/anchore_engine/services/policy_engine/engine/policy/bundles.py
+++ b/anchore_engine/services/policy_engine/engine/policy/bundles.py
@@ -1111,9 +1111,9 @@ class ExecutableBundle(VersionedEntityMixin):
 
                     for policy_id in rule.policy_ids:
                         if len(policies[policy_id]) > 1:
-                            raise DuplicateIdentifierFoundError(identifier=rule.policy_id, identifier_type='policy')
+                            raise DuplicateIdentifierFoundError(identifier=policy_id, identifier_type='policy')
                         if not policies[policy_id]:
-                            raise ReferencedObjectNotFoundError(reference_id=rule.policy_id, reference_type='policy')
+                            raise ReferencedObjectNotFoundError(reference_id=policy_id, reference_type='policy')
 
                         self.policies[policy_id] = ExecutablePolicy(policies[policy_id][0], strict_validation=strict_validation)
 

--- a/anchore_engine/services/policy_engine/engine/policy/exceptions.py
+++ b/anchore_engine/services/policy_engine/engine/policy/exceptions.py
@@ -126,7 +126,7 @@ class ValidationError(PolicyError):
 
 class ReferencedObjectNotFoundError(ValidationError):
     def __init__(self, reference_type, reference_id):
-        super().__init__('Referenced bundle object not found')
+        super().__init__('Referenced object not found')
         self.reference_type = reference_type
         self.reference_id = reference_id
 

